### DISCUSSION TITLE
how to execute a query with timeout?

### DISCUSSION THREAD

**Discussion by luoliyou** [godror/godror#268] [Q&A]
when I execute a query, the oracle server maybe 'hang', so my client will keep waiting......
I try to use queryContext(),but it keep waiting, too.
so, How do I set parameter, for example, Does godror support the parameter that named 'socket timeout'?

**Comment by tgulacsi** [comment]
```
ctx, cancel := context.WithTimeout(context.Background(), 4*time.Second)
_, err := db.QueryRowContext(ctx, "SELECT COUNT(0) FROM all_objects")
cancel()
```
should work.

What have you tried?

**Reply by luoliyou** [reply]
I have try your way, the  problem is that, when 'Oracle-server'  is hang , the timeout I set will be invalid, the method does not return  deadlineExceeded.
